src/Storage/Doctrine/PhpcrOdm/MetadataSubscriber.php

```diff
@@ -38,9 +41,45 @@ public function getSubscribedEvents()
     {
         return [
             Event::loadClassMetadata,
+            Event::prePersist,
         ];
     }
 
+
+    /**
+     * TODO: Put this in a dedicated subscriber.
+     */
+    public function prePersist(LifecycleEventArgs $args)
+    {
+        $document = $args->getObject();
+        $metadataFactory = $args->getObjectManager()->getMetadataFactory();
+        $odmMetadata = $metadataFactory->getMetadataFor(ClassUtils::getRealClass(get_class($document)));
+
+        if (null === $ctMetadata = $this->metadataFactory->getMetadataForClass($odmMetadata->getName())) {
+            return;
+        }
+
+        $documentId = $odmMetadata->getIdentifierValue($document);
+
+        foreach ($odmMetadata->childrenMappings as $childrenField) {
+            if (!isset($ctMetadata->propertyMetadata[$childrenField])) {
+                continue;
+            }
+
+            $children = $odmMetadata->getFieldValue($document, $childrenField);
+
+            foreach ($children as $index => $child) {
+                $childMetadata = $metadataFactory->getMetadataFor(ClassUtils::getRealClass(get_class($child)));
+                $childMetadata->setIdentifierValue($child, sprintf(
+                    '%s/cmfct:%s-%s',
```

Comment:
need to encode the field name in order to prevent conflicts, for instance if we have 2 properties, `slidershow` and `slideshow-big` then the filter will return wrong results.